fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/PartialDataV2RequestHandler.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.handlers;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery;
+import com.yahoo.bard.webservice.web.DataApiRequest;
+import com.yahoo.bard.webservice.web.responseprocessors.ResponseProcessor;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * A request handler that builds responses which filter partial data V2.
+ */
+public class PartialDataV2RequestHandler implements DataRequestHandler {
+
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private final DataRequestHandler next;
+
+    /**
+     * Constructor.
+     *
+     * @param next  Next Handler in the chain
+     */
+    public PartialDataV2RequestHandler(@NotNull DataRequestHandler next) {
```

Comment:
Perhaps `DruidBackedPartialDataRequestHandler`? That indicates that it's using Druid for at least part of it's job. The other important aspect is to make sure what is going on, and why, is clear from the class's JavaDoc